homeassistant/components/light/limitlessled.py

```diff
@@ -271,128 +225,72 @@ def rgb_color(self):
-        self.group.temperature = 0.0
-        self._color = WHITE
-        self.group.brightness = 0.0
-        self._brightness = _to_hass_brightness(1.0)
-        self._temperature = _to_hass_temperature(self.group.temperature)
-        self.group.on = False
-
-    @property
-    def rgb_color(self):
-        """Return the color property."""
-        return self._color
-
-    @property
-    def color_temp(self):
-        """Return the temperature property."""
-        return self._temperature
-
-    @property
-    def supported_features(self):
-        """Flag supported features."""
-        return SUPPORT_LIMITLESSLED_RGBWW
+        if args:
+            pipeline.transition(transition_time, **args)
 
-    # pylint: disable=arguments-differ
-    @state(True)
-    def turn_on(self, transition_time, pipeline, **kwargs):
-        """Turn on (or adjust property of) a group."""
-        from limitlessled.presets import COLORLOOP
-        # Check arguments.
-        if ATTR_BRIGHTNESS in kwargs:
-            self._brightness = kwargs[ATTR_BRIGHTNESS]
-        if ATTR_RGB_COLOR in kwargs:
-            self._color = kwargs[ATTR_RGB_COLOR]
-        elif ATTR_COLOR_TEMP in kwargs:
-            self._temperature = kwargs[ATTR_COLOR_TEMP]
-        # White is a special case.
-        if min(self._color) > 256 - RGB_BOUNDARY:
-            pipeline.white()
-            self._color = WHITE
-        # Set up transition.
-        if self._color == WHITE:
-            pipeline.transition(
-                transition_time,
-                brightness=_from_hass_brightness(self._brightness),
-                temperature=_from_hass_temperature(self._temperature)
-            )
-        else:
-            pipeline.transition(
-                transition_time,
-                brightness=_from_hass_brightness(self._brightness),
-                color=_from_hass_color(self._color)
-            )
         # Flash.
-        if ATTR_FLASH in kwargs:
+        if ATTR_FLASH in kwargs and self._supported & SUPPORT_FLASH:
             duration = 0
             if kwargs[ATTR_FLASH] == FLASH_LONG:
                 duration = 1
             pipeline.flash(duration=duration)
+
         # Add effects.
-        if ATTR_EFFECT in kwargs:
+        if ATTR_EFFECT in kwargs and self._supported & SUPPORT_EFFECT:
             if kwargs[ATTR_EFFECT] == EFFECT_COLORLOOP:
+                from limitlessled.presets import COLORLOOP
                 self.repeating = True
                 pipeline.append(COLORLOOP)
             if kwargs[ATTR_EFFECT] == EFFECT_WHITE:
                 pipeline.white()
                 self._color = WHITE
 
-
-def _from_hass_temperature(temperature):
-    """Convert Home Assistant color temperature units to percentage."""
-    return 1 - (temperature - 154) / 346
-
-
-def _to_hass_temperature(temperature):
-    """Convert percentage to Home Assistant color temperature units."""
-    return 500 - int(temperature * 346)
+    def _from_hass_temperature(self):
```

Comment:
Why's this conversion function nested in `turn_on`, while the others aren't?

Reply:
It is not nested in `turn_on` but I moved it from global to the class because it now uses `self.min_mireds`/`self.max_mireds` rather than the 154/500 constants from Hue.

The inconsistency is a good catch, though. I will clean up the remaining conversion functions before merging.

Thanks for reviewing!